### DISCUSSION TITLE
Beta 0.7

### DISCUSSION THREAD

**Discussion by not-nef** [not-nef/txt2#63] [Announcements]
New Features:
- The scrollbar and the footer are back
- The window can be resized now
- Markdown preview is fixed
- New Start page

<hr /><em>This discussion was created from the release <a href='https://github.com/not-nef/onyx/releases/tag/v0.7'>Beta 0.7</a>.</em>

**Comment by not-nef** [comment]
@rdbende can you give me a professional opinion on how readable my code is?

**Reply by rdbende** [reply]
Professional :) I like it.
And, yeah can give some opinion, though I wouldn't call myself professional.

Overall I think the code is readable, the only thing that makes it hard to read are the statements squeezed into one line

https://github.com/not-nef/onyx/blob/1f4665d320efc3632431d4d8fa9a87f126350626/src/main.py#L22-L23

https://github.com/not-nef/onyx/blob/1f4665d320efc3632431d4d8fa9a87f126350626/src/tabmanager.py#L96-L97

I used to write them like that too, but I realized that actually I don't save that many lines, however, when I read it later, it's much harder to just look at it and understand immediately what that part does.
There's an amazing tool called [`black`](https://github.com/psf/black), that detects things like this and automatically formats your code, and formats it in a way that really makes it easier to understand at a glance what the code is doing.

Otherwise what I found a bit smelly is the use of global variables, and the `vars` module.
I don't want to explain here why global variables are considered evil (in any programming language, actually), but in Onyx the global variables and the `vars` module could be replaced by object oriented programming in all places. Besides, OOP is much more logical and intuitive and thus makes it much easier for humans to **understand** the code.


> [shameless plug]
I wrote a tool, that helps you detect problems and potential errors in your Tkinter program, and it also recommends good coding styles with Tkinter. You might find it useful for Onyx. [flake8-tkinter ->](https://github.com/rdbende/flake8-tkinter)

**Reply by not-nef** [reply]
Thanks a lot! will try out black right now. im a little bit scared of oop since i dont fully understand it so im gonna stick to global variables for a bit longer.

**Comment by arieltordiglione** [comment]


**Reply by rdbende** [reply]
It was not him. It was a spambot.

**Reply by arieltordiglione** [reply]
"I am punishing you by ignoring you, ghost. I hope you can feel the pain."
its a strange phrase hahaha

**Reply by lg-nineoneninethreenine** [reply]
Yeah. I don't know how this BS even works here.

**Comment by tushantydv17** [comment]
what the fcuk is this, nd why I'm in the CC.
________________________________

**Reply by lg-nineoneninethreenine** [reply]
Well, that was the result of a large-scale attack. It has happened everywhere, so it is common rn.

**Comment by not-nef** [comment]
# TO EVERYONE THAT WANTS TO KNOW WHY THEY GOT MENTIONED

This Discussion fell victim to a spambot who mentioned a few dozens of people. I am terribly sorry for the inconvenience. Have a nice day!

**Comment by lg-nineoneninethreenine** [comment]
It was the result of a large-scale attack. We aren't alone w. r. t. the comments.
